dyno-demo/src/main/java/com/netflix/dyno/demo/redis/DynoJedisDemo.java

```diff
@@ -292,16 +294,32 @@ public void runScanTest(boolean populateKeys) throws Exception {
 
 		logger.info("Reading keys from {} with pattern {}", this.clusterName, keyPattern);
 		CursorBasedResult<String> cbi = null;
+		long start = System.currentTimeMillis();
+		int count = 0;
 		do {
-			cbi = client.dyno_scan(cbi, 10000, keyPattern);
+			try {
+
+				cbi = client.dyno_scan(cbi, 5, keyPattern);
+			} catch (PoolOfflineException ex) {
+				logger.info("Caught exception.... retrying scan");
+				cbi = null;
+				continue;
+			}
+
 
 			List<String> results = cbi.getStringResult();
+			count += results.size();
+			int i = 0;
 			for (String res : results) {
-				logger.info(res);
+				logger.info("{}) {}", i, res);
+				i++;
 			}
-		} while (!cbi.isComplete());
+			Thread.sleep(1000);
```

Comment:
You are trying to calculate how fast SCAN performs but at the same time you seem to be sleeping the thread in the loop.

Reply:
man. I was doing some testing and I left this. This is not really doing performance testing but gives more time for say taking a node down, etc